dbms/src/Functions/FunctionsStringReplace.h

```diff
@@ -217,13 +153,76 @@ class FunctionStringReplace : public IFunction
                 ErrorCodes::ILLEGAL_COLUMN);
     }
 
+    void executeImplConstHaystack(
+        const ColumnPtr & column_src,
+        const ColumnPtr & column_needle,
+        const ColumnPtr & column_replacement,
+        bool needle_const,
+        bool replacement_const,
+        ColumnWithTypeAndName & column_result) const
+    {
+        auto res_col = ColumnString::create();
+        res_col->reserve(column_src->size());
+
+        RUNTIME_CHECK_MSG(
+            !needle_const || !replacement_const,
+            "should got here when all argments of replace is constant");
```

Comment:
you mean "should not got here when all arguments of replace is constant"?

Reply:
yes typo